### DISCUSSION TITLE
C++ extension intellisense using gcc not g++

### DISCUSSION THREAD

**Discussion by minerva-firmware** [microsoft/vscode-discussions#289] [Extension Development QnA] [answered]
I've assign a "new" file extension type to c++, by using Settings->Text Editor->Files mapping "*.xx" to "cpp". Turning on intellisense debug logging I see that the now file is being analysed by the intellisense extension:

  define: __ATOMIC_RELEASE=3
  define: __declspec(x)=__attribute__((x))
  define: _DEBUG
  define: UNICODE
  define: _UNICODE
  **_other: --gcc
  other: --gnu_version=110300
  stdver: c17_**
  intelliSenseMode: linux-gcc-x64
Queueing IntelliSense update for files in translation unit of: E:\mfcl_engineering\workspace-eclipse\mfc_axis_example\src\mfc_axis_example.xx

But the debug log above shows that C is being used as the analysis language not C++. Is this a bug in intellisense?

**Comment by isidorn** [comment, marked as answer]
This is a good question for the C++ extension here https://github.com/microsoft/vscode-cpptools
